### DISCUSSION TITLE
GraphQL API and default schema

### DISCUSSION THREAD

**Discussion by paragliding-nsk** [torchbox/wagtail-grapple#375] [Q&A] [answered]
Hello, again I am asking for help from professionals. Am I right that in wagtail-grapple the default schema file only supports two models: Page and StreamField? And in order to display custom models through the GraphQL API, one need to either create a custom schema file or edit the default schema? So far all the options I've found using the default schema are much more complex than using a custom schema file, but perhaps I haven't found the best options?

**Comment by zerolab** [comment, marked as answer]
@paragliding-nsk Grapple will generate schema for any supported models. Here's what it looks at by default: https://github.com/torchbox/wagtail-grapple/blob/main/grapple/registry.py#L50-L57

**Reply by paragliding-nsk** [reply]
Thanks for the tip, can you also clarify: “django models” means a model inherited from django.db.models? And where can I find the most up-to-date documentation? It looks like this one:
https://wagtail-grapple.readthedocs.io/en/latest/general-usage/decorators.html
is a little outdated, at least some fields are not imported correctly (Eg: 'from wagtail.admin.edit_handlers import FieldPanel')

**Reply by zerolab** [reply]
>  “django models” means a model inherited from django.db.models
That is correct.

> https://wagtail-grapple.readthedocs.io/en/latest/general-usage/decorators.html
is a little outdated, at least some fields are not imported correctly (Eg: 'from wagtail.admin.edit_handlers import FieldPanel')

ah, I knew we missed something. Opened https://github.com/torchbox/wagtail-grapple/issues/376
Other than imports, the docs should be in an OK place (famous last words, eh?)
